tracer/src/Datadog.Trace/SpanExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="SpanExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Extension methods for the <see cref="ISpan"/> interface
+    /// </summary>
+    public static class SpanExtensions
+    {
+        /// <summary>
+        /// Sets the details of the user on the local root span
+        /// </summary>
+        /// <param name="span">The span to be tagged</param>
+        /// <param name="userDetails">The details of the current logged on user</param>
+        public static void SetUser(this ISpan span, UserDetails userDetails)
```

Comment:
nit: as this is a public "consumer facing" method, should we add a null check to both of these arguments?

Reply:
Good idea. Do we have a policy around throwing on null vs ignoring and returning?

Reply:
> Do we have a policy around throwing on null vs ignoring and returning

I wouldn't have said _policy_. Generally, we shouldn't break customers if they didn't do anything. But I think throwing null is ok here as they're explicitly calling an API with an invalid value. _Better_ to fail fast in that case IMO. 

> This is similar to the decision in #2437